e2e/main_test.go

```diff
@@ -205,6 +222,62 @@ func (s *FinalityContractTestSuite) Test4SubmitFinalitySignature() {
 	// submit finality signature
 	err = s.ExecuteContract(s.contractAddr, fp.Address(), contractMsgJson)
 	s.NoError(err)
+
+	// query to confirm signature is stored
+	queryVoters := QueryMsg{
+		BlockVoters: &BlockVoters{
+			Height:  startHeight,
+			HashHex: fmt.Sprintf("%x", blockToVote.AppHash),
+		},
+	}
+	queryVotersJson, err := json.Marshal(queryVoters)
+	s.NoError(err)
+	votersResBz := s.QueryContract(s.contractAddr, string(queryVotersJson))
+	var votersRes BlockVotersResponse
+	err = json.Unmarshal(votersResBz, &votersRes)
+	s.NoError(err)
+	s.Equal(1, len(votersRes))
+	s.Equal(fpBTCPK.MarshalHex(), votersRes[0].FpBtcPkHex)
+}
+
+func (s *FinalityContractTestSuite) Test4_SubmitFinalitySigDuplicate() {
```

Comment:
I don't think we need to test this in e2e because it is an edge case that does not involve Babylon layer